internal/provider/resource_apply_module.go

```diff
@@ -359,13 +430,13 @@ func (t *applyModuleResource) Delete(ctx context.Context,
 	}
 }
 
-// doRun launches a remote run and waits for it to complete.
-func (t *applyModuleResource) doRun(ctx context.Context,
-	input *doRunInput, output *doRunOutput,
```

Comment:
Any reason the output is being passed as a param here instead of returning it as a return value?

Reply:
That's how the function was originally designed and coded prior to this bug fix story.  The function has been renamed to createRun but its signature was otherwise left as originally coded.  Looking more closely, it seems to be the Terraform Way for methods to return only diagnostics as return values and to pass both input and output structs by pointer.  I recall having seen lots of functions in the Terraform framework that follow that pattern.  Please inform if you want this story to redesign the signature of that function.

Reply:
Lets update this to return the output struct (same for the other function in this module that takes in an output as a param)